packages/next/next-server/lib/i18n/detect-locale-cookie.ts

```diff
@@ -1,7 +1,7 @@
 import { IncomingMessage } from 'http'
 
 export function detectLocaleCookie(req: IncomingMessage, locales: string[]) {
-  if (req.headers.cookie && req.headers.cookie.includes('NEXT_LOCALE')) {
+  if (req.headers.cookies && req.headers.cookies.hasOwnProperty('NEXT_LOCALE')) {
```

Comment:
`req.headers.cookies` is not a thing in `IncomingMessage`.

```suggestion
  if (req.headers.cookie && req.headers.cookie.hasOwnProperty('NEXT_LOCALE=')) {
```